packages/puppeteer-core/src/common/types.ts

```diff
@@ -29,6 +29,13 @@ export interface Moveable {
   move(): this;
 }
 
+/**
+ * @internal
+ */
+export interface Disposed {
```

Comment:
Can we/should we create a `super` interface that extend `Disposable`,`AsyncDisposable`,`Disposed`, that way we can be more sure of that the decorator would work in most cases?

Reply:
Since decorators wrap things, using decorators as a means of composition is considered better than inheritance (see https://www.digitalocean.com/community/tutorials/composition-vs-inheritance).

Reply:
Discussed offline.